dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/utils/McpClient.java

```diff
@@ -0,0 +1,145 @@
+                              "version": "1.0.0"
+                            }
+                          }
+                        }
+                        """;
+
+        HttpRequest httpRequest = new HttpRequest(server, path + "/mcp").requestProp("Accept", ACCEPT_HEADER)
+                                                                        .requestProp(MCP_PROTOCOL_HEADER, MCP_PROTOCOL_VERSION)
+                                                                        .requestProp("Mcp-Session-Id", sessionId)
+                                                                        .jsonBody(request)
+                                                                        .method("POST");
+        String response = httpRequest.run(String.class);
+
+        String expectedResponse = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": "1",
+                          "result": {
+                            "protocolVersion": "2025-06-18",
+                          }
+                        }
+                        """;
+        JSONAssert.assertEquals(expectedResponse, response, JSONCompareMode.LENIENT);
+
+        sessionId = httpRequest.getHeader("Mcp-Session-Id");
+        assertNotNull(sessionId);
+
+        // Notify the server that initialization was successful
+        String notification = """
+                         {
+                           "jsonrpc": "2.0",
+                           "method": "notifications/initialized"
+                         }
+                        """;
+
+        callMCPNotification(server, path, notification);
+    }
+
+    @Override
+    protected void after() {
+        try {
+            new HttpRequest(server, path + "/mcp").requestProp("Mcp-Session-Id", sessionId)
+                                                  .method("DELETE")
+                                                  .run(String.class);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String callMCP(LibertyServer server, String path, String jsonRequestBody) throws Exception {
+        return new HttpRequest(server, path + "/mcp")
+                                                     .requestProp("Accept", ACCEPT_HEADER)
+                                                     .requestProp(MCP_PROTOCOL_HEADER, MCP_PROTOCOL_VERSION)
+                                                     .requestProp("Mcp-Session-Id", sessionId)
+                                                     .jsonBody(jsonRequestBody)
+                                                     .method("POST")
+                                                     .run(String.class);
+    }
+
+    public void callMCPNotification(LibertyServer server,
+                                    String path,
+                                    String jsonRequestBody)
+                    throws Exception {
```

Comment:
Please add Javadoc.

The corresponding methods on `HttpTestUtils` already have some.